sandbox/libs/authn/src/main/java/org/opensearch/authn/Permission.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.authn;
+
+/**
+ * A class that defines a Permission.
+ *
+ * Example "opensearch.indexing.index.create"
+ *
+ * @opensearch.experimental
+ */
+public class Permission {
+
+    public final String PERMISSION_DELIMITER = "\\.";
+
+    public final String ACTION_DELIMITER = "/";
+
+    public String permissionString;
+
+    public String[] permissionSegments;
+
+    public String resource;
+
+    public String action;
+    public String permissionType;
+
+    public Permission(String permission) {
+
+        this.permissionString = permission;
+        this.permissionSegments = permissionString.split(PERMISSION_DELIMITER);
```

Comment:
This looks prone to IndexOutOfBoundsException if the permission is not formatted as expected.

Reply:
Oof good catch.